src/main/java/org/omnifaces/util/JNDIObjectLocator.java

```diff
@@ -360,7 +223,221 @@ private InitialContext createInitialContext() {
 	 * This deals with transient final fields correctly.
 	 */
 	private Object readResolve() {
-		return builder.build();
+		return toBuilder().build();
 	}
 
+        /*
+                The following are derived from de-lomboked builder pattern objects for this class
+        */
+        @ConstructorProperties({"environment", "portableNamePrefix", "noCaching", "cacheRemote"})
+        JNDIObjectLocator(final Map<String, String> environment, final String portableNamePrefix, final boolean noCaching, final boolean cacheRemote) {
+                this.environment = environment;
+                this.namespace = portableNamePrefix;
+                this.noCaching = noCaching;
+                this.cacheRemote = cacheRemote;
+		initialContext = new Lazy<>(this::createInitialContext);
+		initialContextLock = new ReentrantLock();
+		jndiObjectCache = new Lazy<>(() -> this.noCaching ? emptyMap() : new ConcurrentHashMap<>());
+		remoteAnnotation = new Lazy<>(() -> toClassOrNull("javax.ejb.Remote"));
+        }
+
+        public static class JNDIObjectLocatorBuilder {
+                private ArrayList<String> environmentKey;
+                private ArrayList<String> environmentValue;
```

Comment:
Should be declared against interface. 
Variable names should be plural.
Why not a Map<String, String> by the way?

Reply:
This was delomok'd code. I didn't do all this by hand :)

Reply:
I'll change it, et's easy